### PR TITLE
Create Scheduler that runs peridically

### PR DESCRIPTION
**Pull request by alexjoybc**
this PR includes the following proposed changes:
* a `QuartzHostedService` to run the different jobs in the background
* Configuration and documentation of the `SearchRequestJob`

### CODE REVIEW

File: app/docker-compose.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ version: '3.7'
 services:
   search-api:
     build:
-      context: .
+      context: SearchApi
```
Comment (pdittaro) [inline app/docker-compose.yml]: If possible, I'd suggest moving the docker-compose file to the parent directory. Reason being is that we may introduce other components of the fams3 stack that won't fit in the `app` folder. Plus, I think its a common practice to put it at the top level directory.

Comment (alexjoybc) [reply app/docker-compose.yml]: agree

File: app/DynamicsAdapter/DynamicsAdapter.Web/Controllers/WeatherForecastController.cs [hunk 2]
```diff
@@ -1,7 +1,6 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
-using System.Threading.Tasks;
```
Comment (peggy-gov-bc-ca) [inline app/DynamicsAdapter/DynamicsAdapter.Web/Controllers/WeatherForecastController.cs]: Do we really need this file? If yes, can we change the file name?

File: app/DynamicsAdapter/DynamicsAdapter.Web/Infrastructure/QuartzHostedService.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        }
+
+        /// <summary>
+        /// When the hosted service stops, it shutdowns all the services.
+        /// </summary>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public async Task StopAsync(CancellationToken cancellationToken)
+        {
+            await Scheduler?.Shutdown(cancellationToken);
```
Comment (peggy-gov-bc-ca) [inline app/DynamicsAdapter/DynamicsAdapter.Web/Infrastructure/QuartzHostedService.cs]: is it possible that Scheduler is null?

File: app/DynamicsAdapter/DynamicsAdapter.Web/Controllers/WeatherForecastController.cs (hunk 2 above)
Comment (alexjoybc) [reply app/DynamicsAdapter/DynamicsAdapter.Web/Controllers/WeatherForecastController.cs]: this should not be part of this pr lol, this controller is going to go away anyway, it's just the default controller provided by the boilerplate..

File: app/DynamicsAdapter/DynamicsAdapter.Web/Infrastructure/QuartzHostedService.cs (hunk 3 above)
Comment (alexjoybc) [reply app/DynamicsAdapter/DynamicsAdapter.Web/Infrastructure/QuartzHostedService.cs]: yes it is

File: app/DynamicsAdapter/DynamicsAdapter.Web/Controllers/WeatherForecastController.cs (hunk 2 above)
Comment (pdittaro) [reply app/DynamicsAdapter/DynamicsAdapter.Web/Controllers/WeatherForecastController.cs]: I'm alright with removing this as part of the PR

File: docker-compose.yml [hunk 4]
```diff
@@ -17,6 +17,21 @@ services:
...
+      JAEGER_SERVICE_NAME: dynamics-adapter
+      JAEGER_SAMPLER_TYPE: const
+      JAEGER_ENDPOINT: http://jaeger-collector:14268/api/traces
+      SCHEDULER__CRON: 0/5 * * * * ?
+    restart: always
+    networks:
+      - search-api-net
     
   rabbitmq:
     image: rabbitmq:3.7.15-management
```
[the remark this answers is not in the document]
Comment (alexjoybc) [reply docker-compose.yml]: yup

File: app/DynamicsAdapter/DynamicsAdapter.Web/Controllers/WeatherForecastController.cs (hunk 2 above)
Comment (alexjoybc) [reply app/DynamicsAdapter/DynamicsAdapter.Web/Controllers/WeatherForecastController.cs]: not doing anything for this in this PR

**Pull request merged**